src/lib/list/selection-list.ts

```diff
@@ -97,6 +98,11 @@ export class MatSelectionListChange {
     'tabindex': '-1',
     '[class.mat-list-item-disabled]': 'disabled',
     '[class.mat-list-item-with-avatar]': '_avatar || _icon',
+    // Manually set the "primary" or "warn" class if the color has been explicitly
+    // set to "primary" or "warn". The pseudo checkbox picks up these classes for
+    // its theme. The accent theme palette is the default and doesn't need to be set.
+    '[class.mat-primary]': 'color === "primary"',
+    '[class.mat-warn]': 'color === "warn"',
```

Comment:
I'm sure we could argue whether we also want to have the `mat-accent` class here. I'm fine doing that if desired, but it's _currently_ not necessary.

Also we could set the classes in the setter instead of having to maintain that list in the `host` bindings (it's just the most convenient and Angular idiomatic way)

Reply:
Wouldn't this also end up styling any of the other content that goes into the list item? Alternatively we could put the class on the parent of the checkbox.

Reply:
Not sure. Isn't this what we always do? e.g. `<mat-toolbar>` which also has projected content.

Reply:
The CSS should generally be defined as e.g. `.mat-toolbar.mat-primary`

Reply:
Exactly. so this should be fine.

Reply:
I've generally avoided adding bindings for the theme classes since they tend to be static and it adds three (or two here) bindings for one property. I don't feel super strongly, though.

Reply:
Yeah, I remember that reasoning. I don't feel too strong either. It's just more overhead to manually manage the class/handle state updates from the parent `MatSelectionList`.